src/main/scala/gitbucket/core/controller/ControllerBase.scala

```diff
@@ -371,6 +371,23 @@ trait AccountManagementControllerBase extends ControllerBase {
     }
   }
 
+  protected def uniqueExtraMailAddress(paramName: String = ""): Constraint = new Constraint() {
+    override def validate(
+      name: String,
+      value: String,
+      params: Map[String, Seq[String]],
+      messages: Messages
+    ): Option[String] = {
+      getAccountByMailAddress(value, true)
+        .filter { x =>
+          if (paramName.isEmpty) true else Some(x.userName) != params.optionValue(paramName)
+        }
+        .map { _ =>
+          "Mail address is already registered."
+        }
```

Comment:
@takezoe getAccountByMailAddress checks primary and extra mailaddresses. so, same mail address is denied when across different user. But, this `validate` accept same mail address in same user's primary/extra fields. It can be checked by javascript. Can I use such javascript?

Reply:
OK, I wrote check logic in `validate`